qiskit/quantum_info/synthesis/qsd.py

```diff
@@ -94,13 +116,15 @@ def qs_decomposition(mat, opt_a1=True, decomposer_1q=None, decomposer_2q=None):
         else:
             circ.ucry((2 * vtheta).tolist(), qr[:-1], qr[-1])
         # right circ
-        right_circ = _demultiplex(u1, u2, opt_a1=opt_a1)
+        right_circ = _demultiplex(u1, u2, opt_a1=opt_a1, opt_a2=opt_a2, _depth=_depth)
         circ.append(right_circ.to_instruction(), qr)
 
+    if opt_a2 and _depth == 0:
+        return _apply_a2(circ)
     return circ
 
 
-def _demultiplex(um0, um1, opt_a1=False):
+def _demultiplex(um0, um1, opt_a1=False, opt_a2=False, _depth=0):
```

Comment:
We should add `opt_a2` and `_depth` to the docstring. I guess pylint doesn't complain because this is a private method, but it'd be better to document what these do.

Reply:
maybe? mentioned above